### PR TITLE
fix: datatype number for search

### PR DESCRIPTION
**Pull request by MyvTsv**
- [x] I have performed a self-review of my code.
- [x] I have added tests (when available) that prove my fix is effective or that my feature works.

## Description

- It fixes !35167
- Change the decimal type of number fields to display the decimal part of the decimal numbers in search options

## Screenshots (if appropriate):
Capture d’écran du 2024-11-18 14-11-57

### CODE REVIEW

**Comment by stonebuzz** [comment]
Does searching with `<` still work?

**Comment by MyvTsv** [comment]
> Does searching with `<` still work?

Yes the operator is functional

Capture d’écran du 2024-11-18 15-15-19
Capture d’écran du 2024-11-18 15-15-11

**Review by stonebuzz — APPROVED** [review]

**Comment by stonebuzz** [comment]
Can you propose a patch to the user to get another validation that it works properly?

**Review by Rom1-B — CHANGES_REQUESTED** [review]
Seems OK, waiting customer feedback.

**Review by Rom1-B — APPROVED** [review]

**Comment by stonebuzz** [comment]
@MyvTsv can you adapt ``CHANGELOG.md`

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ## [UNRELAESE]
 
 ### Fixed
+- Change the decimal type of number fields to display the decimal part of the decimal numbers in search options.
```
Comment (stonebuzz) [inline CHANGELOG.md:11]:
```suggestion
- Force decimal ‘datatype’ of `numeric` fields  for more accurate display.
```

**Review by stonebuzz — CHANGES_REQUESTED** [review]

**Review by stonebuzz — APPROVED** [review]

**Pull request merged**